### PR TITLE
Feature/issue2521

### PR DESCRIPTION
**Pull request by salomon-j**

### CODE REVIEW

File: grails-app/assets/javascripts/meriplan.js [hunk 1]
```diff
@@ -404,7 +405,18 @@ function MERIPlan(project, projectService, config) {
     self.removeAsset = function(asset) {
         self.meriPlan().assets.remove(asset);
     };
-
+    self.addControl = function () {
```
Comment (chrisala) [inline grails-app/assets/javascripts/meriplan.js:408]: I don't think that add/remove row makes sense for the control approach - it's basically a yes/no with reasons.

File: grails-app/assets/javascripts/meriplan.js [hunk 2]
```diff
@@ -638,6 +650,8 @@ function DetailsViewModel(o, project, budgetHeaders, risks, config) {
     self.threats = new GenericViewModel(o.threats, ['threat', 'intervention']);
     self.consultation = ko.observable(o.consultation);
     self.communityEngagement = ko.observable(o.communityEngagement);
+    self.controlApproach = new GenericViewModel(o.controlApproach, ['data1', 'data2']);
```
Comment (chrisala) [inline grails-app/assets/javascripts/meriplan.js:653]: Despite some of the original sections using this, don't use 'data1' / 'data2' as field names as it makes the data not very useful in other contexts (e.g Download and API JSON is pretty meaningless if all of the fields don't have descriptive names).  See "self.threats" for an example of how to use the GenericViewModel with field names.

File: grails-app/assets/javascripts/meriplan.js [hunk 3]
```diff
@@ -329,6 +329,8 @@ function MERIPlan(project, projectService, config) {
     self.protectedNaturalAssests = ['Natural/Cultural assets managed', 'Threatened Species', 'Threatened Ecological Communities',
         'Migratory Species', 'Ramsar Wetland', 'World Heritage area', 'Community awareness/participation in NRM', 'Indigenous Cultural Values',
         'Indigenous Ecological Knowledge', 'Remnant Vegetation', 'Aquatic and Coastal systems including wetlands', 'Not Applicable'];
+    self.controls = ['Yes', 'No'];
+    self.methodType = ['Natural', 'Chemical', 'Disease Virus', 'Mechanical', 'Genetics', 'Other with comment field'];
```
Comment (chrisala) [inline grails-app/assets/javascripts/meriplan.js:333]: Move this methodType drop down into the configuration JSON I think as it's likely we'll be asked to change it.  Or maybe leave it here but allow it to be overwritten?

File: grails-app/assets/javascripts/meriplan.js (hunk 2 above)
Comment (chrisala) [inline grails-app/assets/javascripts/meriplan.js:653]: Rename "self.controlApproach" to "self.threatToNativeSpecies" as that is what is actually being recorded.

**Review by chrisala — CHANGES_REQUESTED** [review]

File: grails-app/assets/javascripts/meriplan.js (hunk 1 above)
Comment (salomon-j) [reply grails-app/assets/javascripts/meriplan.js:408]: I've noticed just now on the latest doc file that Blaz sent yesterday that she also updated this section, she removed the option to add rows.

Comment (salomon-j) [reply grails-app/assets/javascripts/meriplan.js]: I've pushed the changes for this

File: grails-app/assets/javascripts/meriplan.js (hunk 2 above)
Comment (salomon-j) [reply grails-app/assets/javascripts/meriplan.js]: I've pushed the changes for this

File: grails-app/assets/javascripts/meriplan.js (hunk 3 above)
Comment (salomon-j) [reply grails-app/assets/javascripts/meriplan.js]: I've pushed the changes for this

File: grails-app/assets/javascripts/meriplan.js (hunk 2 above)
Comment (salomon-j) [reply grails-app/assets/javascripts/meriplan.js]: I've pushed the changes for this

**Review by chrisala — APPROVED** [review]

**Review by chrisala — CHANGES_REQUESTED** [review]
Thanks Joseph, these changes are fine however I've noticed you've templated the heading in meriPlan/_programOutcome.gsp but not meritPlanReadOnly/_programOutcome.gsp

**Review by chrisala — APPROVED** [review]

**Pull request merged**